api/v1beta1/ociclusteridentity_types.go

```diff
@@ -0,0 +1,102 @@
+/*
+ Copyright (c) 2022, 2023 Oracle and/or its affiliates.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+type PrincipalType string
+
+const (
+	// UserPrincipal represents a user principal.
+	UserPrincipal PrincipalType = "UserPrincipal"
+)
+
+// OCIClusterIdentitySpec defines the parameters that are used to create an OCIClusterIdentity.
+type OCIClusterIdentitySpec struct {
+	// Type is the type of OCI Principal used.
+	// UserPrincipal is the only supported value
+	Type PrincipalType `json:"type"`
+
+	// PrincipalSecret is a secret reference which contains the authentication credentials for the principal.
+	// +optional
+	PrincipalSecret corev1.SecretReference `json:"clientSecret,omitempty"`
```

Comment:
Should `clientSecret` be `principalSecret`?

Reply:
fixed